lib/code_corps/github/event/installation/repos.ex

```diff
@@ -7,76 +7,155 @@ defmodule CodeCorps.GitHub.Event.Installation.Repos do
     GithubAppInstallation,
     GithubRepo,
     GitHub.Installation,
+    GitHub,
     Repo
   }
 
   alias CodeCorps.GitHub.Adapters.GithubRepo, as: GithubRepoAdapter
 
   alias Ecto.{Changeset, Multi}
 
-  @typep outcome :: {:ok, list(GithubRepo.t)} |
-                    {:error, CodeCorps.GitHub.api_error_struct} |
-                    {:error, :invalid_repo_payload}
+  @typep repo_result :: {:ok | :error, GithubRepo.t | Changeset.t}
+  @typep aggregated_result :: {:ok | :error, list(GithubRepo.t | Changeset.t)}
+
+  @doc ~S"""
+  Marks a `GithubAppInstallation` as "processing" and immediately returns it.
+
+  In the background, it fires a task to asynchronously call &process/1
+  """
+  @spec process_async(GithubAppInstallation.t) :: {:ok, GithubAppInstallation.t, Task.t}
+  def process_async(%GithubAppInstallation{} = installation) do
+    {:ok, %GithubAppInstallation{} = processing_installation} = installation |> set_state("processing")
+
+    task = Task.Supervisor.async(:background_processor, fn -> processing_installation |> process() end)
+
+    {:ok, processing_installation, task}
```

Comment:
Returning the task is mostly in the service of  testing. We can use `Task.await/1` to resolve it in tests.